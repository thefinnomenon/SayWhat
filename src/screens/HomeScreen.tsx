import React from 'react';
import { Theme } from '../../types';
import styled from 'styled-components/native';
import { useNavigation } from 'react-navigation-hooks';
import { useTranslation } from 'react-i18next';
import Button from '../components/Button';
import normalize from '../../responsive';

export const HomeScreen = () => {
  const { navigate } = useNavigation();
  const { t } = useTranslation();

  return (
    <Container>
      <Spacer />
      <Logo source={require('../../assets/images/logo.png')} />
      <Spacer />
      <ButtonsContainer>
        <StyledButton onPress={() => navigate('Decks')} title={t('PLAY')} />
        <StyledButton onPress={() => navigate('Rules')} title={t('RULES')} />
        <StyledButton
          onPress={() => navigate('Settings')}
          title={t('SETTINGS')}
        />
      </ButtonsContainer>
      <Spacer />
    </Container>
  );
};

const Container = styled.View<Theme>`
  flex: 1;
  background: ${props => props.theme.colors.background};
`;

const Logo = styled.Image<Theme>`
  height: ${normalize(250)};
  width: ${normalize(250)};
  margin: auto;
  resize-mode: contain;
`;

const ButtonsContainer = styled.View`
  height: ${normalize(200)};
  width: ${normalize(300)};
  justify-content: space-between;
  align-content: center;
  margin: auto;
`;

const StyledButton = styled(Button)<Theme>`
  margin: ${normalize(6)}px;
`;

const Spacer = styled.View`
  flex: 1;
`;

HomeScreen.navigationOptions = {
  header: null,
};

export default HomeScreen;
