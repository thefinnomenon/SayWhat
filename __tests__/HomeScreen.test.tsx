import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from '../src/screens/HomeScreen';

var module = require('react-navigation-hooks');
module.useNavigation = jest.fn().mockImplementation(() => {
  return {
    navigate: jest.fn(),
  };
});

it('renders correctly', () => {
  renderer.create(<HomeScreen />);
});
