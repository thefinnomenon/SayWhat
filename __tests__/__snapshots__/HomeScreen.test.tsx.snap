// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/logo.png",
      }
    }
    style={
      Array [
        Object {
          "height": 500,
          "marginBottom": "auto",
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": "auto",
          "resizeMode": "contain",
          "width": 500,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "height": 400,
          "justifyContent": "space-between",
          "marginBottom": "auto",
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": "auto",
          "width": 600,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignContent": "center",
          "backgroundColor": "#50b7ff",
          "borderRadius": 20,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 40,
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }
      >
        PLAY
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignContent": "center",
          "backgroundColor": "#50b7ff",
          "borderRadius": 20,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 40,
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }
      >
        RULES
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignContent": "center",
          "backgroundColor": "#50b7ff",
          "borderRadius": 20,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 40,
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }
      >
        SETTINGS
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
</View>
`;
